src/sources/csv.rs

```diff
@@ -0,0 +1,264 @@
+//! CSV source for syncing with Famedly's Zitadel.
+
+use std::{fs::File, path::PathBuf};
+
+use anyhow::{anyhow, Context, Result};
+use async_trait::async_trait;
+use csv::Reader;
+use serde::Deserialize;
+
+use super::Source;
+use crate::{config::FeatureFlags, user::User, zitadel::SourceDiff, Config, FeatureFlag};
+
+/// CSV Source
+pub struct CsvSource {
+	/// CSV Source configuration
+	csv_config: CsvSourceConfig,
+	/// Optional set of features
+	feature_flags: FeatureFlags,
+	/// IDP ID
+	idp_id: String,
+}
+
+#[async_trait]
+impl Source for CsvSource {
+	fn get_name(&self) -> String {
+		"CSV".into()
+	}
+
+	async fn get_diff(&self) -> Result<SourceDiff> {
+		let new_users = self.read_csv()?;
+		return Ok(SourceDiff { new_users, changed_users: vec![], deleted_user_ids: vec![] });
+	}
+}
+
+impl CsvSource {
+	/// Create a new CSV source
+	pub fn new(config: &Config) -> Result<Self> {
```

Comment:
```rust
pub fn new(config: &CsvSourceConfig) -> Result<Self>
```
for two reasons:
- `CsvSource` doesn't have to access the stuff that is not needed for it (i.e. other sources configs)
- `CsvSource::new` should be called only if `CsvSourceConfig` is present. Failing inside with `CSV configuration is missing` is 🚩

Reply:
Done.